src/query/functions/aggregation/take.go

```diff
@@ -49,9 +50,17 @@ func NewTakeOp(
 		return baseOp{}, fmt.Errorf("operator not supported: %s", opType)
 	}
 
-	heap := utils.NewFloatHeap(takeTop, int(params.Parameter))
-	fn := func(values []float64, buckets [][]int) []float64 {
-		return takeFn(heap, values, buckets)
+	var fn takeFunc
+	k := int(params.Parameter)
+	if k < 1 {
+		fn = func(values []float64, buckets [][]int) []float64 {
+			return takeNone(values, buckets)
+		}
+	} else {
+		heap := utils.NewFloatHeap(takeTop, k)
```

Comment:
What if k >2?

Reply:
Same thing as if k == 2? Don't understand the question

Reply:
sorry, misread the code..